### ISSUE TITLE
Remove `pudl_settings` and `pudl_setup` in favor of PUDL_INPUT and PUDL_OUTPUT

### ISSUE THREAD

**Issue by jdangerx** [catalyst-cooperative/pudl#2293]
```[tasklist]
### Scope
- [x] pudl_setup command is gone
- [x] pudl_settings object is not used anymore
- [x] "Creating a workspace" portion of development docs are updated
- [x] `pudl-examples` notebooks work
```

```[tasklist]
### Next steps
- [x] go through list that @bendnorman put in the comments and start removing references to pudl_settings.
```

**Comment by bendnorman** [comment]
To deprecate pudl_settings and pudl_setup we'll need to remove all references to `pudl_settings` and `pudl.workspace.setup.get_defaults()`. Locations I could find:

- [ ] `pudl.analysis.service_territory`:gets default pudl_settings and creates a SQL engine to pass to `PudlTabl`. It also passes pudl_settings to `pudl.output.censusdp1tract.get_layer()`.
- [ ] `pudl.analysis.state_demand`: loads default and creates SQL engine.
- [ ] `pudl.convert.censusdb1tract_to_sqlite`: gets sqlite output path from pudl_settings. Also gets the defaults to initializing a datastore.
- [ ] `pudl.convert.datasette_metadata_to_yml`: passes pudl_settings to `DatasetteMetadata.from_data_source_ids(pudl_settings=pudl_settings)`.
- [ ] `pudl.convert.epacems_to_parquet`: will mostly be handled by #2292
- [ ] `pudl.extract.ferc1`: Mostly references will be removed and functionality has already been converted to dagster.
- [ ] `pudl.extract.ferc714`: References pudl_settings but I don't actually think it uses it?
- [ ] `pudl.glue.ferc1_eia`: `get_plants_ferc1_raw` uses pudl_settings but I'm not sure what the function is actually used for. Looks like it's just used for an integration test @zaneselvans?
- [ ] `pudl.metadata.classes`: DatasetteMetadata.from_data_source_ids() and Datasource.add_datastore_metadata() need pudl_settings to construct a datastore.
- [ ] `pudl.output.censusdptract`: gets defaults and passes to `censusdp1tract_to_sqlite`
- [ ] `pudl.output.epacems`: gets cems path from defaults
- [ ] `pudl.output.ferc714`: Just passes pudl_settings to pudl.output.censusdp1tract.get_layer()
- [ ] `pudl.output.pudlabl`: Passes pudl_settings to etl_ferc714 to construct a datastore for the extract method.
- [ ] `pudl.transform`: passes pudl_settings to the extract methods that have been replaced by dagster concepts.
- [ ] `pudl.workspace.setup`:  million references that don't matter because we can delete this whole file!


We mostly need to  replace the pudl_settings object with reading paths from the `PUDL_OUTPUT` and `PUDL_CACHE` env vars.

Ripping out pudl_settings would introduce another large set of changes into the dagster-asset-etl branch. An alternative would be to keep pudl_settings in this iteration, merge the big PR in then open a new PR that makes all of the pudl_settings changes. To do this we'd have to create a pudl_settings dagster resource instead of using the env vars. What do y'all think @zaneselvans @zschira @jdangerx?

**Comment by zschira** [comment]
I'm definitely in favor of removing `pudl_settings`, but I think ripping it out after the big merge would probably smart. I lean towards doing the bare minimum to get a functioning and tested implementation into `dev`, then doing all the nice cleanup things in small future PR's.

**Comment by jdangerx** [comment]
Thanks for doing all this digging @bendnorman - that will definitely be useful once we are ready to pull the settings out! I agree with @zschira, I think we should wait on this until the big PR is merged, so that we have smaller/more manageable changes.

**Comment by zaneselvans** [comment]
@bendnorman I thought `pudl.glue.ferc1_eia.get_plants_ferc1_raw()` was used during the PUDL plant ID mapping process to generate the list of new plan IDs that need to be mapped (in the manual association of FERC & EIA plants that we do each year as new FERC plants appear, or old ones appear in new guises, so that we can connect the two datasets, even though there's no shared ID). But I'm not seeing the script or any notebooks that use it now, so I'm not sure where that code has wandered off to.

**Comment by bendnorman** [comment]
@zaneselvans that's strange. @cmgosnell are you familiar with the `pudl.glue.ferc1_eia.get_plants_ferc1_raw()` function?

@jdangerx @zschira I agree with y'all! I'll create a PR replacing the env vars with a `pudl_settings` resource.

**Comment by cmgosnell** [comment]
`pudl.glue.ferc1_eia.get_plants_ferc1_raw()` is used as a part of the [annual plant and utility id mapping process](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/pudl_id_mapping.html#checking-for-unmapped-records).

I added it as a test because its pretty easy to run in this setup and I made it into a tox env called `get_unmapped_ids`

**Comment by zschira** [comment]
Ah ok I can work on converting this to use dagster, as I've been working on getting these tests working

**Comment by bendnorman** [comment]
@cmgosnell I'm a little confused how `get_plants_ferc1_raw()` is used in the mapping process. Do you just run it in a notebook when doing the manual mapping process? I'm curious because it's the only function that uses the extract_dbf and extract_xbrl functions in the dagster branch. Could we convert the `get_plants_ferc1_raw()` to a dagster job?

**Comment by cmgosnell** [comment]
> @cmgosnell I'm a little confused how get_plants_ferc1_raw() is used in the mapping process. Do you just run it in a notebook when doing the manual mapping process? I'm curious because it's the only function that uses the extract_dbf and extract_xbrl functions in the dagster branch. Could we convert the get_plants_ferc1_raw() to a dagster job?

This function is run as a part of the prep for the FERC-EIA manual mapping process. the skinny of it is that we need *all* of the FERC plants that are going to be loaded into the db. You can't load the FERC and EIA data into a pudl without using `ignore-foreign-key-constraints`.  There are several long steps in the ETL process that we can effectively avoid by using this baby mapping-only "ET" - the most time consuming being the FERC plant-id-er which is a part of the steam table's transform process.

This function does a few things:
* run a baby FERC extract step with only the plant tables.
* run a baby transform step using `GenericPlantFerc1TableTransformer`, which basically only does the transformations necessary to get the plant name strings as they will be attempted to be loaded into `pudl.sqlite`
* Poops out a table with plants and utils de-duplicated and in the format we need for checking against the existing manually compiled glue

The bb extract step here is the only thing that is actually using the settings object because that was previously necessary.

How its actually run:
I've been running this via the `tox -e get_unmapped_ids`. This test runs this FERC-side ET and an EIA only ETL w/o any FK constraints. It checks the plants/utils we expect to get based on the prepped data against what is currently in `pudl/package_data/glue/pudl_id_mapping.xlsx` and saves CSVs in `devtools/ferc1-eia-glue` of any missing plants or utils that we would need to then manually add them into the mapping sheets.

iirc i put this `glue_test.py` into the integration bucket because it can be run against the fast sqlite db in the integration tests. So in the integration test context it just checks if the last two years of ferc and eia data are compatible with the manually compiled IDs.

we could probably run this function as a dag.... but tbqh i'm still not totally sure what the design constraints are there/what is needed. But maybe we should just chat about it because i'm sure its possible!

**Comment by zschira** [comment]
I created #2303 for tracking this. I added a proposed plan for converting this process to dagster, but let me know if I'm missing something/this plan doesn't make sense.

**Comment by zaneselvans** [comment]
I suspect there's probably still a fair bit of fiddling required to finish this transition, including getting rid of the `pudl_setup` command entirely and ensuring that everything that refers to the settings files is getting them directly from `src/pudl/package_data/settings/etl_{full,fast}.yml` but we should do it and wrap this up!
